src/js/settings.js

```diff
@@ -506,7 +509,42 @@ function importSubscriptions(){
     })
   });
 }
+/**
+* Import NewPipe Channel Subscriptions
+* @return {Void}
+*/
+function importNewpipeSubscriptions(data){
 
+  progressView.seen = true;
+  progressView.width = 0;
+    
+  let newpipe, n, link, newpipesub, counter;
+      
+      newpipe = JSON.parse(data);
+      counter = 0;
+      
+      for (n in newpipe.subscriptions) {
+
+          link = newpipe.subscriptions[n].url.split("/");
+  
+          invidiousAPI('channels', link[4], {}, (data)=> {
+            newpipesub = {
+              channelId: data.authorId,
+              channelName: data.author,
+              channelThumbnail: data.authorThumbnails[2].url
+            };
+            counter++;
+              addSubscription(newpipesub, false);
+              progressView.progressWidth = (counter / newpipe.subscriptions.length) * 100;
+    });
+  }
+  if ((counter + 1) == newpipe.subscriptions.length) {
```

Comment:
Add this if statement into the invidiousAPI callback.  Node runs asynchronously so the statement is ran before the last API call is finished.